test/wasm_test.cc

```diff
@@ -195,6 +196,13 @@ TEST_P(TestVm, AlwaysApplyCanary) {
             auto wasm_handle_comp =
                 createWasm(vm_key, source, plugin_comp, wasm_handle_factory_comp,
                            wasm_handle_clone_factory_for_canary, false);
+            // Validate that canarying is cached for the first baseline plugin variant.
+            if (first) {
+              first = false;
+              EXPECT_EQ(canary_count, 0);
```

Comment:
I'm clearly missing something, since the tests pass, but shouldn't it be the other way around (i.e. `canary_count == 1` when canary is performed the first time, and `canary_count == 0` when it's cached)?

Reply:
No you have it right, but there's another call above the loop that makes the first iteration to be cached and others not cached.

Reply:
Oh, thanks, I completely missed that.